File: app/src/main/java/com/zl/weilu/androidut/bean/Banana.java
package com.zl.weilu.androidut.bean;

/**
 * @Description:
 * @Author: weilu
 * @Time: 2017/11/18 11:54.
 */

public class Banana extends Fruit{

    private static String COLOR = "黄色的";

    public Banana() {}

    public static String getColor() {
        return COLOR;
    }

    public String getBananaInfo() {
        return flavor() + getColor();
    }

    private String flavor() {
        return "甜甜的";
    }

    public final boolean isLike() {
        return true;
    }
}

File: app/src/main/java/com/zl/weilu/androidut/bean/Banana.kt
package com.zl.weilu.androidut.bean

/**
 * @Description:
 * @Author: weilu
 * @Time: 2017/11/18 11:54.
 */

object Banana : Fruit() {

    val bananaInfo: String
        get() = flavor() + COLOR

    val isLike: Boolean
        get() = true


    private fun flavor(): String {
        return "甜甜的"
    }

    private val COLOR = "黄色的"

    @JvmStatic
    fun color(): String{
        return COLOR
    }

}
